src/Simplex/Chat/Library/Commands.hs

```diff
@@ -4163,6 +4171,39 @@ agentSubscriber = do
 
 type AgentSubResult = Map ConnId (Either AgentErrorType (Maybe ClientServiceId))
 
+runRelationsVectorMigration :: CM ()
+runRelationsVectorMigration = do
+  liftIO $ threadDelay' 5000000 -- 5 seconds (initial delay)
+  migrateUserNotAdminGroups
+  migrateMembers
+  where
+    stepDelay = 100000 -- 0.1 second
+    migrateUserNotAdminGroups = flip catchAllErrors eToView $ do
+      lift waitChatStartedAndActivated
+      gIds <- withStore' getUserNotAdminGroupIds
+      forM_ gIds $ \gId -> do
+        withStore' (\db -> setEmptyVectorForGroupId db gId) `catchAllErrors` eToView
+        liftIO $ threadDelay' stepDelay
+    migrateMembers = flip catchAllErrors eToView $ do
+      lift waitChatStartedAndActivated
+      gmIds <- withStore' getGMsWithoutVectorIds
+      forM_ gmIds $ \gmId -> do
+        processMember gmId `catchAllErrors` eToView
+        liftIO $ threadDelay' stepDelay
+      unless (null gmIds) migrateMembers
+      where
+        processMember gmId =
+          withGroupMemberLock "runRelationsVectorMigration" gmId $
+            withStore' (\db -> getMemberRelationsVector_' db gmId) >>= \case
+              Just _ -> pure () -- already has vector
+              Nothing -> do
+                introducedRelations <- withStore' $ \db -> getIntroducedRelations db gmId
+                introducedToRelations <- withStore' $ \db -> getIntroducedToRelations db gmId
+                connectedRelations <- withStore' $ \db -> getIntroConnectedRelations db gmId
```

Comment:
why these are three queries and not one?

Reply:
I would try to pack it all in a single query avoiding the need to read members completely, as the change is not transactional.

Reply:
and we will need to make this migration anyway, so maybe we should start from it, and here use the same function just for one member.